addons/account_peppol/models/account_move.py

```diff
@@ -21,6 +21,19 @@ class AccountMove(models.Model):
         copy=False,
         readonly=True,
     )
+    ref = fields.Char(compute='_compute_payment_reference', readonly=False, store=True)
+
+    def _compute_payment_reference(self):
+        # EXTENDS account
+        super()._compute_payment_reference()
+        for move in self.filtered(lambda m: (
+            m.state != 'cancelled'
```

Comment:
I think we should also filter out moves that have a `peppol_move_state`, because that means that the XML file has been already generated?